riscv/processor.cc

```diff
@@ -393,6 +393,44 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   const reg_t henvcfg_init = (proc->extension_enabled(EXT_SVPBMT) ? HENVCFG_PBMTE : 0);
   csrmap[CSR_HENVCFG] = henvcfg = std::make_shared<henvcfg_csr_t>(proc, CSR_HENVCFG, henvcfg_mask, henvcfg_init, menvcfg);
 
+  const reg_t mstateen0_mask = (proc->extension_enabled(EXT_ZFINX) ? MSTATEEN0_FCSR : 0) |
+                               MSTATEEN0_CS | MSTATEEN0_HENVCFG | MSTATEEN_HSTATEEN;
```

Comment:
All these masks should be calculated in the class constructors, such as `mstatush_csr_t::mstatush_csr_t()` does.

Reply:
Why? I want to use a general class for all mstateen csrs. If we caculated the masks in the class constructor, we should use different class for different mstateen since they have different mask.

Reply:
Yeah, It's OK to calculate reg mask this way to reuse the code. My question is whether is necessary to calculate mask in the constructor just like mstatush_csr_t::mstatush_csr_t() does.

Reply:
It's not strictly "necessary" but if it's not a big burden then it's better to put all the details of the CSR into one place (its class) instead of spreading them out. But see my other comment about calculating masks; it might make more sense to leave this here.